src/app/[locale]/clientSearchDataset.ts

```diff
@@ -0,0 +1,25 @@
+import {SearchOptions, SearchResult} from '@/lib/dataset-fetcher';
+
+interface SearchDatasets {
+  licenses: string[];
+  publishers: string[];
+}
+
+// only use this function for client components
+// For server component use the dataset-fetcher directly
+export default async function clientSearchDatasets({
```

Comment:
Would it be an option to not use `default` here? This allows for future change: if we want to add another function to this file, we don't have to update the exports (i.e. the API is stable).